survey_mailer/survey_mailer.go

```diff
@@ -0,0 +1,198 @@
+		bodyHtml += strings.Join(attendees, "<br />")
+		bodyHtml += "</p>"
+	}
+	if len(sendingErrors) > 0 {
+		bodyText += "The following addresses did not receive the email due to sending errors: "
+		bodyText += strings.Join(sendingErrors, ", ")
+		bodyText += ". "
+		bodyHtml += "<p>The following addresses did not receive the email due to sending errors <br />"
+		bodyHtml += strings.Join(sendingErrors, "<br />")
+		bodyHtml += "</p>"
+	}
+
+	if bodyText != "" {
+		sendEmail(to, subject, bodyText, bodyHtml)
+		log.Println("Sending email of missing emails and errors.")
+	}
+}
+
+func sendEmail(to string, subject string, bodyText string, bodyHtml string) error {
+	from := config.SurveyFromEmail
+	bodyHtml += `<br /><img src="https://adb.dxe.io/static/img/logo1.png" height="46" width="50">`
+	// EnvConfig uses the AWS credentials in the environment
+	// variables $AWS_ACCESS_KEY_ID nd $AWS_SECRET_KEY.
+	_, err := ses.EnvConfig.SendEmailHTML(from, to, subject, bodyText, bodyHtml)
+	if err != nil {
+		return errors.New(to)
+	}
+	return nil
+}
+
+func bulkSendEmails(event model.Event, subject string, bodyText string, bodyHtml string) {
+	var sendingErrors []string
+	for _, recipient := range event.AttendeeEmails {
+		log.Println("Sending email to:", recipient)
+		// Send email
+		err := sendEmail(recipient, subject, bodyText, bodyHtml)
+		if err != nil {
+			sendingErrors = append(sendingErrors, err.Error())
+		}
+	}
+	sendMissingEmail(event.EventName, event.AttendeeMissingEmails, sendingErrors)
+}
+
+func updateSurveyStatus(db *sqlx.DB, eventId int) {
+	// Update "survey_sent" to true (1)
+	_, err := model.UpdateEventSurveyStatus(db, model.Event{
+		ID:         eventId,
+		SurveySent: 1,
+	})
+	if err != nil {
+		log.Println("ERROR:", err)
+	}
+}
+
+func survey(db *sqlx.DB, surveyOptions SurveyOptions) {
+	log.Println("Looking for", surveyOptions.SurveyType, "events on", surveyOptions.QueryDate)
+
+	// Get events matching query that that haven't had surveys sent yet
+	events, err := model.GetEvents(db, model.GetEventOptions{
+		DateFrom:       surveyOptions.QueryDate,
+		DateTo:         surveyOptions.QueryDate,
+		EventType:      surveyOptions.QueryEventType,
+		EventNameQuery: surveyOptions.QueryEventName,
+		SurveySent:     "0",
+	})
+	if err != nil {
+		log.Printf("Failed to get events: %v", err)
+		return
+	}
+
+	// Iterate through events
+	for _, event := range events {
+		subject := "Survey: " + event.EventName
+		// set linkParam based on LinkParam option
+		linkParam := ""
+		if surveyOptions.LinkParam == "name" {
+			linkParam = strings.Replace(event.EventName, " ", "+", -1)
+		}
+		if surveyOptions.LinkParam == "date" {
+			linkParam = event.EventDate.Format("2006-01-02")
+		}
+		// build body by replacing LINK_PARAM with the actual link param
+		bodyText := strings.Replace(surveyOptions.BodyText, "LINK_PARAM", linkParam, -1)
+		bodyHtml := strings.Replace(surveyOptions.BodyHtml, "LINK_PARAM", linkParam, -1)
```

Comment:
String replacement into HTML is generally unsafe; e.g., the event name might contain <>.

This isn't a security issue since XSS doesn't apply to HTML email, and maybe we don't allow <> in event names anyway, but I'd probably at least put a `TODO` comment about applying proper escaping.

Reply:
Made a comment for now. Thanks.

Reply:
I am actually using the EscapeString function from here: https://golang.org/pkg/html/

Additionally, the ADB does not let you create event names with ```<```, ```>```, or ```&``` in the event name.